entrypoint.sh

```diff
@@ -39,9 +39,17 @@ fi
 # Function to push a resource to Superblocks if it has changed
 push_resource() {
     local location="$1"
-    # Push only if there are some changes to $location/application.yaml, $location/page.yaml, $location/api.yaml, or $location/apis/*.
-    # This is to avoid pushing when only the components have changed.
-    if echo "$changed_files" | grep -qP "^${location}/(application|page|api).yaml" || echo "$changed_files" | grep -qP "^${location}/apis/" ; then
+
+    # Escape any special characters in the resource subdir
+    escaped_location="${location%%/*}/$(printf '%s\n' "${location#*/}" | sed 's/[][\\^$.|?*+(){}/]/\\&/g')"
```

Comment:
This is producing concatenation of escaped and non-escaped location:

```
➜  ~ location='(ahmet)_test'
➜  ~ escaped_location="${location%%/*}/$(printf '%s\n' "${location#*/}" | sed 's/[][\\^$.|?*+(){}/]/\\&/g')"
➜  ~ echo $escaped_location
(ahmet)_test/\(ahmet\)_test
```
Is that expected?

Reply:
If so, can we add an example escaped case here?

Reply:
yes it is expected. the unescaped prefix (part preceding and inclusive of the first forward slash) is the resource type directory, which is controlled completely by us - `apps/` and `backends/` are the two possibilities specifically.

the suffix is the resource subdir which can contain the special characters. the goal of this form of escaping was to ensure we don't escape the resource type directory path (specifically the first slash) but I decided that the entire prefix was safe to leave unescaped

Reply:
the comment in line 43 specifies what part of the location we are escaping